### DISCUSSION TITLE
Authentication API documentation

### DISCUSSION THREAD

**Discussion by andriyndev** [serverpod/serverpod#827] [Ideas]
I think it'd be great to describe in the documentation how to work with authentication API directly to be able to implement custom authentication forms. In the current version of documentation there is "If you are building custom forms, it is helpful to read the documentation provided in the module and review the code in the SignInWithEmailDialog". But I didn't find any detailed documentation about it except of the code, and it's unclear what is the correct way to do it: call methods of EmailAuthController or call caller.email.* functions directly, and how stable it will be (cannot it be renamed in the future that'll break my code). IMHO, as a backend framework, Serverpod should focus on providing API for interacting with server instead of implementing view (of course, the latter is also a plus, but shouldn't be the main option). The current approach is also unsuitable for developers who want to separate view and data layers in their app because the ready-to-use authentication form communicates with the server directly under the hood.

**Comment by vlidholt** [comment]
Great idea. I added an issue to track the progress here: #843

**Reply by andriyndev** [reply]
Thank you!

**Comment by Ifoegbu1** [comment]
@andriyndev  The doc quite pointed to the code for separating the logic from the UI ... In the SignInWithEmailDialog, you'll see the client methods communicating with the server. You then can easily extract it and use in your custom forms:

 `final EmailAuthController _emailAuth =
      EmailAuthController(client.modules.auth);`
`registration({
    required String userName,
    required String fullName,
    required String email,
    required String password,
    required BuildContext context,
  }) async {
    _isLoading = true;
    update();

    try {
      var success =
          await _emailAuth.createAccountRequest(userName, email, password);...`
